vmdb/app/models/ems_kubernetes.rb

```diff
@@ -0,0 +1,21 @@
+class EmsKubernetes < EmsContainer
+  has_many :container_nodes,                      :foreign_key => :ems_id, :dependent => :destroy
+  has_many :container_groups,                       :foreign_key => :ems_id, :dependent => :destroy
+  has_many :container_services,                   :foreign_key => :ems_id, :dependent => :destroy
+
+  def self.ems_type
+    @ems_type ||= "kubernetes".freeze
+  end
+
+  def self.description
+    @description ||= "Kubernetes".freeze
+  end
+
+  def inventory
+    require 'kubeclient'
+    # TODO, timeout params
+    url = "http://#{hostname}:#{port}/api/"
+    client = Kubeclient::Client.new  url, "v1beta3"
+    client.get_all_entities
+  end
```

Comment:
Good call @Fryguy.

@abonas, the gist is to let the URI library build the URL and use the hostname method to automatically handle ipv6 literal addresses.

Something like this:

```ruby
  require 'kubeclient'
  require 'uri'

  # TODO, timeout params
  uri = URI::HTTP.build(:path => "/api", :port => port)

  # URI::Generic#hostname was added in ruby 1.9.3 and will automatically 
  # wrap an ipv6 address in []
  uri.hostname = hostname

  client = Kubeclient::Client.new  uri, "v1beta3"
  client.get_all_entities
```

Note, Kubeclient::Client.new **should** accept a URI object but if not, change 

```ruby
  client = Kubeclient::Client.new  uri, "v1beta3"
```

To:
```ruby
  client = Kubeclient::Client.new  uri.to_s, "v1beta3"
```

See related change here:
https://github.com/ManageIQ/manageiq/pull/1080/files#diff-7804811082dba7a7c91cde72206fd648R48